core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1386,6 +1386,12 @@ abstract class RDD[T: ClassTag](
   }
 
   /**
+   * @return false if and only if the RDD contains no elements at all. Note that an RDD
+   *         may be empty even when it has at least 1 partition.
+   */
+  def nonEmpty(): Boolean = !isEmpty()
```

Comment:
ok, I will close it. but I think it may be useful to add some methods like nonEmpty/takeRight...